core/src/mindustry/world/blocks/storage/Unloader.java

```diff
@@ -61,7 +61,7 @@ public void updateTile(){
                     int pos = (offset + i) % proximity.size;
                     var other = proximity.get(pos);
 
-                    if(other.interactable(team) && other.block.unloadable && other.block.hasItems
+                    if(other.interactable(team) && other.block.unloadable && other.unloadable() && other.block.hasItems
```

Comment:
press to implement the above comment, not sure if this will break on blocks with no building entity or something?

```suggestion
                    if(other.interactable(team) && other.unloadable() && other.block.hasItems
```